### ISSUE TITLE
"Redefinition of module X" errors in 0.38.1

### ISSUE THREAD

**Issue by staminajim** [CocoaPods/CocoaPods#3886]
After updating to CocoaPods 0.38.1 we're now getting a very long list of `Redefinition of module 'AFNetworking'` etc errors.  It looks like any pods that are used by multiple targets in the Podfile are trying to define their module map multiple times, which Xcode doesn't like.

Looks like this new feature: https://github.com/CocoaPods/CocoaPods/issues/3874 is the likely culprit, I'm not sure what the solution is yet.

**Comment by joridor** [comment]
Having the same issue with Restkit.
Downgrading to v 0.38 as a temporary workaround.

**Comment by Rom184** [comment]
Having the same issue with Parse.

**Comment by sarsonj** [comment]
Having same issue with some modules, getting xCode compilation error Cyclic dependency for module 'UIKit:' - Darwin -> UIKit -> MyModule -> UIKit.

**Comment by neonichu** [comment]
Just to clarify, are you all using integration via static libraries? If yes, @staminajim is probably right about the cause of this.

**Comment by staminajim** [comment]
@neonichu Correct.  We still have to support iOS 7 so we can't use dynamic frameworks yet.

**Comment by jiashuw** [comment]
Same here.

**Comment by sebcode** [comment]
Same here. Here is a minimal test project to reproduce the problem:

https://github.com/sebcode/cocoapods-issue-3886

`pod _0.38.1_ install` -> Build -> Errors as described above (Redefinition of module 'AFNetworking')
`pod _0.37.2_ install` -> Build -> No errors

**Comment by segiddins** [comment]
This will be fixed by https://github.com/CocoaPods/CocoaPods/issues/3898, we're aware of it, so please verify that branch works for y'all instead of piling onto this issue

**Comment by segiddins** [comment]
Closed by https://github.com/CocoaPods/CocoaPods/pull/3898.

**Comment by danpe** [comment]
To run the workaround:
```gem install cocoapods -v 0.38.0```
```pod _0.38.0_ install```

**Comment by peymankh** [comment]
I still seem to be having this issue as of 0.38.2

**Comment by neonichu** [comment]
@peymankh you're probably seeing a different issue than this, because the root cause of it was reverted completely. Could you please open a new issue with all the required information, see our [contribution guidelines](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

**Comment by baoluo** [comment]
Hello all, @neonichu I can confirm that I still have this issue, even with the latest beta (0.39 beta 4) 
In order to make it lint can't use the module_map property.

**Comment by neonichu** [comment]
This issue was never about the `module_map` property, please open a new one with more details.

**Comment by pwlkania** [comment]
Hello.
I experienced this issue again with 0.39.0.rc.1. Can anyone confirm?
Tested 0.38.2 - works as expected.
Integration via static libraries (have to support iOS 7).

**Comment by neonichu** [comment]
@pwlkania if you do, please open a new issue with details on how to reproduce

**Comment by pwlkania** [comment]
@neonichu I digged deeper and noticed that I have imported AFNetworking in two places, in the bridging header and in one of the obj-c classes. I removed it from that obj-c header and it started working!
No need to create new issue, 0.39.0.rc.1 works fine in this matter.

**Comment by wonderffee** [comment]
I encountered this problem also after update CocoaPods to 0.39.0. Too many redefinition error for AFNetworking, SDWebImage and so on.
It's strange that when I downgrade CocoaPods to 0.38.2 or 0.37.2, the problem still exist.
Maybe I found the reason of redefinition: take AFNetworking for example, in pch file I use "#import <AFNetworking/AFNetworking.h>", and in another file I use "#import "AFNetworking.h"",  after removing "#import "AFNetworking.h"" the redefinition problem is gone, but only for AFNetworking.
For a variety of reasons, my project may has lots of similar code. It's frustrating to fix all of them.
Everything is well before update, I don't known why. Maybe something is made more strict?
